codec/std/keys.go

```diff
@@ -0,0 +1,141 @@
+package std
+
+import (
+	"errors"
+
+	"github.com/tendermint/tendermint/crypto"
+	"github.com/tendermint/tendermint/crypto/ed25519"
+	"github.com/tendermint/tendermint/crypto/multisig"
+	"github.com/tendermint/tendermint/crypto/secp256k1"
+	"github.com/tendermint/tendermint/crypto/sr25519"
+)
+
+//TODO: Deprecate this file when tendermint 0.34 is released
+
+// PubKeyToProto takes crypto.PubKey and transforms it to a protobuf Pubkey
+func PubKeyToProto(k crypto.PubKey) (PublicKey, error) {
+	var kp PublicKey
+	switch k := k.(type) {
+	case sr25519.PubKeySr25519:
+
+		kp = PublicKey{
```

Comment:
can we use `PublicKey.Set()` (in case it's supported)

Reply:
not supported that is why we need the switch. later on when tendermint defines the types as slices it will be

Reply:
Having a hard time following the goal of this PR. What does this PR accomplish or buy us apart from just moving some stuff over? Until we can work with general slices, what is the point here?

Why not just move over the keys and use interface type and call `Set` and grandfather in multi-sigs?